python_modules/dagster/dagster/_core/execution/context/compute.py

```diff
@@ -281,17 +281,181 @@ def partition_time_window(self) -> TimeWindow:
         """
         return self._step_execution_context.partition_time_window
 
+    @public
+    def has_tag(self, key: str) -> bool:
+        """Check if a logging tag is set.
+
+        Args:
+            key (str): The tag to check.
+
+        Returns:
+            bool: Whether the tag is set.
+        """
+        return self._step_execution_context.has_tag(key)
+
+    @public
+    def get_tag(self, key: str) -> Optional[str]:
+        """Get a logging tag.
+
+        Args:
+            key (tag): The tag to get.
+
+        Returns:
+            Optional[str]: The value of the tag, if present.
+        """
+        return self._step_execution_context.get_tag(key)
+
+    def has_events(self) -> bool:
+        return bool(self._events)
+
+    def consume_events(self) -> Iterator[DagsterEvent]:
+        """Pops and yields all user-generated events that have been recorded from this context.
+
+        If consume_events has not yet been called, this will yield all logged events since the beginning of the op's computation. If consume_events has been called, it will yield all events since the last time consume_events was called. Designed for internal use. Users should never need to invoke this method.
+        """
+        events = self._events
+        self._events = []
+        yield from events
+
+    @public
+    def log_event(self, event: UserEvent) -> None:
```

Comment:
@alangenfeld revisiting the naming here. I think `log_event` is misleading since it implies that it *immediately* writes which it does not.

```python
context.log.info("foo")
context.log_event(...)
context.log.info("bar)
```

prints "foo", then "bar", then the event at at end, which I think it very counterintuitive.

Reply:
this PR is just moving things around, the blame rev for `log_event` is https://github.com/dagster-io/dagster/pull/6304/

i've been opting for `report_`  so think `report_event` would make sense here if want to go through the deprecation dance